### ISSUE TITLE
Testnet Deactivates BSC Network

### ISSUE THREAD

**Issue by Hoanib** [AlphaWallet/alpha-wallet-ios#4556]
When you activate Testnet being BSC as well Ethereum activated on Enabled Networks and switch back to Mainnet BSC appears disengaged.

**Comment by hboon** [comment]
@Hoanib as in:

(A)
1. Ethereum mainnet + BSC
2. Switch to testnet, enable Ropsten
3. Switch to mainnet, BSC is not enabled

(B)
1. Ethereum mainnet + BSC
2. Switch to testnet, enable Ropsten
3. Tap back button
4. Switch to mainnet, BSC is not enabled

(A) should work. (B) doesn't. There's no particular reason why (B) shouldn't work that way except it's extra work.

Are you referring to (A) or (B)?

If it's (A), it's a bug. If it's (B), is there a reason why it should work that way?

**Comment by Hoanib** [comment]
A. Actually it disengages not only BSC when we switch to Mainnet again. The only networks re-engaged are Ethereum, Gnosis and Polygon.

**Comment by hboon** [comment]
@Hoanib can you help verify one more time? The behavior you are referring to sounds like B, not A.

**Comment by Hoanib** [comment]
I switched from Mainnet (wallet connected on Ethereum) to Testnet in Advanced Enabled Networks. Then I clicked on Mainnet Tab and all the current engaged networks on Mainnet were disengaged except Ethereum, Gnosis an Polygon.

**Comment by Hoanib** [comment]
![image](https://user-images.githubusercontent.com/100842739/167540736-830c4280-b424-41a5-bd11-75a70afe04c7.png)

![image](https://user-images.githubusercontent.com/100842739/167540769-bc977d5b-af58-4e6e-a192-57d85cefa549.png)

![image](https://user-images.githubusercontent.com/100842739/167540798-aa9a6c04-5097-40ed-9f44-c36ac188ef66.jpeg)

![image](https://user-images.githubusercontent.com/100842739/167540821-db7152ac-982c-42f3-b045-80bb686c6d30.png)

**Comment by hboon** [comment]
Ah, it's A and sounds like a bug. I can't reproduce it though. Will check

**Comment by Hoanib** [comment]
Are you testing on Apple Store version? Just enable all networks, switch to Testnet, jump to Wallet and jump back to Settings > Enabled Networks, then switch to Mainnet.

**Comment by hboon** [comment]
So you did hit the back button right? i.e B?

**Comment by Hoanib** [comment]
Yes I clicked on the back button to check Wallet and then jumped to Settings > Enabled Networks. If you stay in the same page after activate Testnet and then switch to Mainnet again the enabled networks will be engaged.

**Comment by hboon** [comment]
Thanks for clarifying. So it is (B). Would you comment on this then:

> (A) should work. (B) doesn't. There's no particular reason why (B) shouldn't work that way except it's extra work.
> 
> Are you referring to (A) or (B)?
> 
> If it's (A), it's a bug. If it's (B), is there a reason why it should work that way?

**Comment by Hoanib** [comment]
Yes but the fact that some networks like Ethereum, Gnosis and Polygon remain engaged and the others not it's a bug to me.
